add warning to NaN columns

murilommen opened the pull request:
details:
- it is a known issue on whylogs ([#1300](https://github.com/whylabs/whylogs/issues/1300)) when segmenting by multiple columns that if one of them has NaNs, the pipeline breaks
- adding a warn log that is more informative to what has happened
- adding a warn for NaT on the `ProfileViews` class as well

murilommen committed:
add warning to NaN columns

details:
- it is a known issue on whylogs (#1300) when segmenting
by multiple columns that if one of them has NaNs, the pipeline breaks
- adding a warn log that is more informative to what has happened

murilommen committed:
makes log more informative

murilommen committed:
fix format

naddeoa commented on src/ai/whylabs/templates/batch_bigquery_template.py:
```diff
@@ -147,6 +147,11 @@ def process_batch(self, batch: List[Dict[str, Any]]) -> Iterator[Tuple[SegmentDe
             if len(dataframe) == 0:
                 continue
 
+            if len(self.segment_columns_list) > 1:
+                for col in self.segment_columns_list:
+                    if dataframe[col].isna().values.any():
+                        self.logger.warning(f"Segmenting on column {col} with null values leads to a whylogs KeyError")
+
```
Lets just make it raise if there is going to be an exception.

Also, we can move this up above the for loop and do it once.

murilommen replied:
done

murilommen committed:
Merge remote-tracking branch 'origin/master' into dev/murilommen/warn-segments-on-nans

murilommen committed:
PR comments addressed:
- raising instead of warning
- do one pass on the entire dataframe
- add comment linking to the whylogs issue

murilommen committed:
minor improvements:

- make it raise before other processes
- also warn date grouping on NaTs for unsegmented cases

murilommen committed:
format fix
